ui/pages/multichain-accounts/multichain-account-address-list-page/multichain-account-address-list-page.tsx

```diff
@@ -49,6 +50,28 @@ export const MultichainAccountAddressListPage = () => {
     ? t('receivingAddress')
     : `${accountGroup?.metadata?.name || t('account')} / ${t('addresses')}`;
 
+  // QR Modal state
+  const [isQRModalOpen, setIsQRModalOpen] = useState(false);
+  const [selectedQRData, setSelectedQRData] = useState<{
+    address: string;
+    networkName: string;
+    networkImageSrc?: string;
+  } | null>(null);
+
+  // QR Modal handlers
+  const handleShowQR = useCallback(
+    (address: string, networkName: string, networkImageSrc?: string) => {
+      setSelectedQRData({ address, networkName, networkImageSrc });
+      setIsQRModalOpen(true);
+    },
+    [],
+  );
+
+  const handleCloseQR = useCallback(() => {
+    setIsQRModalOpen(false);
+    setSelectedQRData(null);
+  }, []);
+
```

Comment:
Captures address, network name and image data from the `MultichainAddressRow` and passes it to the `AddressQRCodeModal` as well as handling the open and close state of the `AddressQRCodeModal`